core/rfb.js

```diff
@@ -512,7 +502,9 @@ RFB.prototype = {
         // State change actions
 
         this._rfb_connection_state = state;
-        this.onupdatestate(this, state, oldstate);
+        var event = document.createEvent("CustomEvent");
+        event.initCustomEvent("updatestate", false, false, { state: state });
```

Comment:
we should *really* be using the `CustomEvent` constructor when possible -- the `initCustomEvent` has been deprecated and removed from the standards (https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/initCustomEvent).  With our target browser support, I don't think we can use it all the time, but we can use the polyfill listed here: https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent

Reply:
I was using the constructor to start with, but that went south once I tested on IE. :sob:

I can try the polyfill and see how that goes.